### PR TITLE
Edit learning goals, exam information, prior knowledge

### PR DESCRIPTION
**Pull request by NNieuwenhuis**
Problems:
- delete a skill does not work
- design of adding a skill has a strange behaviour
- all dropdowns in adding a skill take the same value

### CODE REVIEW

File: src/app/components/module-component/edit/module-edit.component.ts [hunk 1]
```diff
@@ -25,17 +29,29 @@ import { Location } from '@angular/common';
...
+  architectural_layers: ArchitecturalLayer[];
 
   //#region selection variables
   selectedLecturer: Lecturer;
   selectedTopic: string;
   private routeSubscription: Subscription;
   selectedTeachingMaterial: string;
   selectedTeachingMaterialType: string;
+  selectedLearningGoal: LearningGoal;
+  selectedArchtecturalLayer: ArchitecturalLayer;
```
Comment (Lehren) [inline src/app/components/module-component/edit/module-edit.component.ts]: typo it should be selectedArchitecturalLayer

**Review by sjoerdbr — APPROVED** [review]

**Review by lukeelten — APPROVED** [review]

File: src/app/models/studentskill.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export interface StudentSkill {
+export class StudentSkill {
```
Comment (lukeelten) [inline src/app/models/studentskill.ts]: INTERFACE!!!

File: src/app/models/learninggoal.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import {StudentSkill} from './studentskill';
 
-export interface LearningGoal {
+export class LearningGoal {
```
Comment (lukeelten) [inline src/app/models/learninggoal.ts]: INTERFACE!!!

File: src/app/models/editmodels/editable_module_output.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import {TeachingMaterial} from '../teaching_material';
 import {Lecturer} from '../lecturer';
+import {LearningGoal} from '../learninggoal';
+import {AssesmentPart} from '../assesment_part';
 import {PriorKnowledgeReference} from '../prior_knowledge_reference.model';
 import {SimpleModule} from '../qualificiationfiltermodels/simple_module';
```
Comment (lukeelten) [inline src/app/models/editmodels/editable_module_output.ts:7]: This should be an interface!!!

File: src/app/models/assesment_part.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-export interface AssesmentPart {
+export class AssesmentPart {
```
Comment (lukeelten) [inline src/app/models/assesment_part.ts]: INTERFACE!!!!

**Review by lukeelten — CHANGES_REQUESTED** [review]
Please make proper use of interfaces to not waste performance on runtime.
There is no need to use a class if you do not want to provide functionality inside.

**Review by lukeelten — APPROVED** [review]

**Pull request merged**